integ-tests/src/main/java/com/google/sql/SqlHelper.java

```diff
@@ -83,4 +87,48 @@ public static void connectAndExecuteQueryAsUser(String username, String password
       throw e;
     }
   }
+
+  /**
+   * Returns count of rows for provided query.
+   *
+   * @param connection DB connection parameter
+   * @param query Select query with count entries names as RowsNR
+   */
+  public static int getEntriesCountForQuery(Connection connection, String query)
+      throws SQLException {
+    int count = 0;
+    try (PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+      ResultSet rs = preparedStatement.executeQuery();
+      rs.next();
+      count = getIntNotNull(rs, "RowsNR");
+    } catch (SQLException e) {
+      LOGGER.error(
```

Comment:
Logging and rethrowing is usually not a good idea. It will likely result in double reporting.

Will this exception not be logged at a higher level anyway?

If you need to preserve context information, rather wrap this exception in an exception with a more detailed error message (and log that higher up).

Reply:
No, so I will remove rethtrowing then

Reply:
right it this case (failing JUnit) should be rethrown